src/components/Item/ItemMenu.vue

```diff
@@ -1,39 +1,45 @@
 <template>
-  <div v-if="items.length > 0">
-    <v-menu
-      absolute
-      close-on-click
-      close-on-content-click
-      :z-index="zIndex"
-      top
-    >
-      <template #activator="{ on, attrs }">
-        <v-btn
-          icon
-          :outlined="outlined"
-          :dark="dark"
-          v-bind="attrs"
-          v-on="on"
-          @click.stop.prevent
-        >
-          <v-icon>mdi-dots-horizontal</v-icon>
-        </v-btn>
-      </template>
-      <v-list dense nav>
-        <v-list-item
-          v-for="(menuItem, index) in items"
-          :key="`item-${item.Id}-menu-${index}`"
-          @click="menuItem.action"
-        >
-          <v-list-item-icon>
-            <v-icon>{{ menuItem.icon }}</v-icon>
-          </v-list-item-icon>
-          <v-list-item-title class="text">
-            {{ menuItem.title }}
-          </v-list-item-title>
-        </v-list-item>
-      </v-list>
-    </v-menu>
+  <div v-if="options.length > 0">
+    <v-fade-transition>
+      <v-menu
+        v-model="show"
+        absolute
+        close-on-click
+        close-on-content-click
+        :z-index="zIndex"
+        :position-x="positionX"
+        :position-y="positionY"
+        top
+      >
+        <template #activator="{ on, attrs }">
+          <v-btn
+            icon
+            :outlined="outlined"
+            :dark="dark"
+            v-bind="attrs"
+            v-on="on"
+            @click.stop.prevent="onActivatorClick"
+            @contextmenu="onRightClick"
```

Comment:
Couldn't we add the `.stop.prevent` here instead of the method?

Reply:
Originally, I planned this to work as follow:

- We get the ref of the parent element
- We bind the element using ``$on`` [docs](https://vuejs.org/v2/api/#vm-on)

That way we could've used that syntax. However, that syntax only works when the event is a custom one (emitted using ``$emit()``) and we're using an standard JS one. This means we need to bind them the JS way, no the Vue way.

That's why the ``e.preventDefault()`` and ``e.stopPropagation()`` are in the event handler.

However, here we're binding the same callback to a ``contextmenu`` event in a child component of the ``ItemMenu`` one (so we can use the SFC syntax, instead of the ref one) in order to reuse the code

We are forced to use the stopPropagation and preventDefault in the function so the paren'ts right click are handled correctly. The decision here relies on if we make another function that gets called for the parent right click or reuse the same code. Something like this:

````
onParentRightClick(e) {
   e.preventDefaukt();
   e.stopPropagation();
   this.onRightClick();
}
onRightClick() {
  // The rest of the current logic. This would be the one that it's referenced in the SFC.
}
````

Reply:
Imo the current approach in this PR is better, the problem is this quirk in Vue's API (which hopefully gets fixed in Vue 3, I haven't checked honestly).

Reply:
Maybe I should add a comment for this?

Reply:
Yes please do :)